### ISSUE TITLE
Question regarding the creation of a new FTP user: change of ownership of all files in public_html to that user 

### ISSUE THREAD

**Issue by pixel-paul** [virtualmin/virtualmin-gpl#374]
Webmin version | 1.984 | Usermin version | 1.834
-- | -- | -- | --
Virtualmin version | 6.17 | Authentic theme version | 19.84.7

When I create a new virtual server, a file is owned by the virtual server owner, in this case for the domain `mytesting.com`:

![Screenshot 2022-02-25 at 12-43-26 (1) _home_mytesting_public_html - File Manager — Webmin 1 984 (Redhat Enterprise Linux 8)](https://user-images.githubusercontent.com/389303/155717366-d245ff53-2d57-46bb-8348-3eb2d979b0af.png)

If I create new FTP users, like so:

![Screenshot 2022-02-25 at 12-44-32 (1) mytesting com - FTP Users — Webmin 1 984 (Redhat Enterprise Linux 8)](https://user-images.githubusercontent.com/389303/155717507-eb5a94e2-5e91-4bbb-aa7a-981d80ddcbd4.png)

The files are now owned by the most recent FTP user created (in this case paul2):

![Screenshot 2022-02-25 at 12-44-55 (1) _home_mytesting_public_html - File Manager — Webmin 1 984 (Redhat Enterprise Linux 8)](https://user-images.githubusercontent.com/389303/155717682-8517f265-2bd8-4c3a-9120-9253633d1642.png)

Is this an expected outcome of creating new FTP users?

**Comment by pixel-paul** [comment]
This also has an odd impact when sudoing to a virtual server owner, as the resultant user is the most recent FTP user created:

```
# su mytesting
[paul2@mytesting.com@test-server root]$
```

**Comment by jcameron** [comment]
This is sort-of expected .. the new FTP user has the same UID as the domain owner, so files may show up as owned by that user because of the way Linux stores file ownership. However, normally they show up as owned by the first occurrence of the UID in /etc/passwd .

**Comment by pixel-paul** [comment]
Ok, understood. Interestingly if a 'normal' user is created, the files stay owned by the virtual server owner and it is still possible to sudo to the virtual server owner.

**Comment by jcameron** [comment]
Yes, a normal user has his own UID so there no conflict.

Are Virtualmin users stored in `/etc/passwd` on your system, or are you using some other user store like LDAP?

**Comment by pixel-paul** [comment]
Yes, they are stored in /etc/passwd.

Is this odd ssh login experience expected? If I try and sudo to the virtual server owner when I have an FTP user created, I am logged in as the FTP user (which is set to /bin/false) - which I believe shouldn't be possible.

**Comment by jcameron** [comment]
Yeah, it's a side-effect of the same issue with the UID lookup. When you're logged in, Linux finds your username from your numeric UID and in this case it's picking the "wrong" entry in /etc/passwd
